src/main/java/edu/kis/powp/jobs2d/command/ComplexCommand.java

```diff
@@ -23,6 +25,19 @@ public void execute(Job2dDriver driver) {
         return commands.iterator();
     }
 
+    @Override
+    public void accept(Visitor visitor) {
+
+    }
+
+    @Override
+    public ICompoundCommand clone() {
+        List<DriverCommand> cloneCommands = new ArrayList<>();
+        this.commands.forEach((e) -> cloneCommands.add(e.clone()));
+
+        return new ComplexCommand(cloneCommands);
+    }
+
```

Comment:
Maybe you implement a default clone method in the ICompoundCommand interface so other implementations won't be obliged to make their own implementations?